### PR TITLE
config_get take testnet dir form env instead of file

### PR DESCRIPTION
**Pull request by NDobrev**

### CODE REVIEW

File: scripts/config-read.shlib [hunk 1]
```diff
@@ -3,13 +3,7 @@ config_read_file() {
...
 
 config_get() {
-    SCRIPT_PATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
-    val="$(config_read_file "${SCRIPT_PATH}"/config.cfg "${1}")";
-    if [ "${val}" = "__UNDEFINED__" ]; then
-        echo -e "Missing config value for ${val}"
-        exit 1
-    fi
-    printf -- "%s" "${val}";
+    printf -- "%s" "${CARDANO_TESTNET_DIR}";
```
Comment (PetarKirov) [inline scripts/config-read.shlib:6]: Instead of hardcoding CARDANO_TESTNET_DIR, this should use bash indirect variable referenes:
https://tldp.org/LDP/abs/html/ivr.html

File: scripts/mkfiles.sh [hunk 2]
```diff
@@ -546,7 +547,7 @@ for NODE in ${BFT_NODES}; do
     echo "  --port                            $(cat ${NODE}/port) \\"
     echo "  --delegation-certificate          ${ROOT}/${NODE}/byron/delegate.cert \\"
     echo "  --signing-key                     ${ROOT}/${NODE}/byron/delegate.key \\"
-    echo "  | tee -a ${ROOT}/${NODE}/node.log"
+    echo "  2>&1 | tee -a ${ROOT}/${NODE}/node.log"
```
Comment (PetarKirov) [inline scripts/mkfiles.sh:550]: why was this change needed?

File: scripts/mkfiles.sh [hunk 3]
```diff
@@ -688,3 +689,7 @@ elif [ "$1" = "shelley" ]; then
 else
   echo "Default yaml configuration applied."
 fi
+
+chmod -R +rw $ROOT
+#find $ROOT -type d -exec chmod 0777 {} \;
+find $ROOT -type f -name "*.*key" -exec chmod 0400 {} \;
```
Comment (PetarKirov) [inline scripts/mkfiles.sh:695]: Why was this need?

File: scripts/mkfiles.sh [hunk 4]
```diff
@@ -108,6 +108,7 @@ $SED -i ${ROOT}/configuration.yaml \
     -e 's/RequiresNoMagic/RequiresMagic/' \
     -e 's/LastKnownBlockVersion-Major: 0/LastKnownBlockVersion-Major: 1/' \
     -e 's/LastKnownBlockVersion-Minor: 2/LastKnownBlockVersion-Minor: 0/'
+chmod a+rw ${ROOT}/configuration.yaml
```
Comment (PetarKirov) [inline scripts/mkfiles.sh:111]: Why was this needed?

File: scripts/automate.sh [hunk 5]
```diff
@@ -36,8 +36,10 @@ kill_running_nodes() {
 
 start_all_nodes() {
   echo "start all the nodes in bg"
-  $ROOT/run/all.sh > /dev/null 2>&1 &
+  pushd $ROOT
+  ./run/all.sh > /dev/null 2>&1 &
   wait_until_count_of_running_nodes 3
+  popd
```
Comment (PetarKirov) [inline scripts/automate.sh:42]: What happens if you don't do `pushd` + `popd`?

**Review by PetarKirov — CHANGES_REQUESTED** [review]

File: scripts/mkfiles.sh (hunk 4 above)
Comment (NDobrev) [reply scripts/mkfiles.sh:111]: because we copied this from nix/store

File: scripts/automate.sh (hunk 5 above)
Comment (NDobrev) [reply scripts/automate.sh:42]: How this is possible?

File: scripts/mkfiles.sh (hunk 3 above)
Comment (NDobrev) [reply scripts/mkfiles.sh:695]: because when we create this folder some of the files/directories are copied from the nix store and have only read permissions.

File: scripts/mkfiles.sh (hunk 2 above)
Comment (NDobrev) [reply scripts/mkfiles.sh:550]: removed

File: scripts/config-read.shlib (hunk 1 above)
Comment (NDobrev) [reply scripts/config-read.shlib:6]: But why?
